models.py

```diff
@@ -17,23 +53,36 @@ def __init__(self, input_vocab_size, config: CfgNode, use_bilstm=False, glove_em
     self.embedding = nn.Embedding(input_vocab_size, config.EMB_DIM)
     self.use_glove = config.GLOVE_EMB_DIM != 0 and glove_embeddings is not None
     self.use_trainable_embeddings = config.EMB_DIM != 0
-
+    # If config.CHAR_EMB_DIM = 0, then the CharacterEmbedding will not be used char_emb from config.
+    self.use_character_level_embeddings = config.CHAR_EMB_DIM != 0
+    self.drop_out = config.ENCODER_DROPOUT_RATE
     if self.use_glove:
       self.glove = nn.Embedding(len(glove_embeddings.keys()), config.GLOVE_EMB_DIM)
       weight_matrix = get_weight_matrix(glove_embeddings, config.GLOVE_EMB_DIM)
       self.glove.load_state_dict({'weight': weight_matrix})
       self.glove.weight.requires_grad = False
-    emb_dim = config.EMB_DIM + config.GLOVE_EMB_DIM if self.use_glove else config.EMB_DIM
+    emb_dim = config.GLOVE_EMB_DIM + config.CHAR_HIDDEN_SIZE if self.use_glove & self.use_character_level_embeddings \
+      else config.GLOVE_EMB_DIM if self.use_glove\
+      else config.CHAR_HIDDEN_SIZE if self.use_character_level_embeddings \
+      else 0
```

Comment:
this is very hard to follow, please only use short ifs if it's something short and concise. You may gain a few lines of code like this, but you lose readability